e2e/__tests__/globalSetup.test.js

```diff
@@ -103,3 +103,22 @@ test('should not call a globalSetup of a project if there are no tests to run fr
   expect(fs.existsSync(project1DIR)).toBe(true);
   expect(fs.existsSync(project2DIR)).toBe(false);
 });
+
+test('should not call any globalSetup if there are no tests to run', () => {
+  const configPath = path.resolve(
+    __dirname,
+    '../global-setup/projects.jest.config.js',
+  );
+
+  const result = runWithJson('global-setup', [
```

Comment:
`runWithJson` actually fails without the fix, but I added the assertions either way to assert they don't run, even if we don't crash anymore

![image](https://user-images.githubusercontent.com/1404810/51910821-8f8c0a00-23d0-11e9-9419-477877f69e93.png)